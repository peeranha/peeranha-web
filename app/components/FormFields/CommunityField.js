import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import arrowDownIcon from 'svg/arrowDown';

import CommunitySelector from 'components/CommunitySelector';
import Img from 'components/Img';
import { Input } from 'components/Input/InputStyled';
import Span from 'components/Span';
import Icon from 'components/Icon';
import IconStyled from 'components/Icon/IconStyled';

import Wrapper from './Wrapper';

const Div = styled.div`
  position: relative;

  ${(props) /* istanbul ignore next */ => Input(props)};

  ${IconStyled} {
    position: absolute;
    right: 0;
  }
`;

export const CommunityField = /* istanbul ignore next */ ({
  input,
  meta,
  label,
  tip,
  disabled,
}) => (
  <Wrapper label={label} tip={tip} meta={meta}>
    <CommunitySelector
      input={input}
      disabled={disabled}
      selectedCommunityId={input.value ? input.value.id : 0}
      Button={({ toggleOpen, communityAvatar, communityLabel }) => (
        <Div
          className="d-flex align-items-center"
          error={meta.touched && (meta.error || meta.warning)}
          disabled={disabled}
          onClick={toggleOpen}
        >
          {communityAvatar && (
            <React.Fragment>
              <Img className="mr-2" src={communityAvatar} alt="comm_img" />
              <Span>{communityLabel}</Span>
            </React.Fragment>
          )}

          <Icon icon={arrowDownIcon} />
        </Div>
      )}
    />
  </Wrapper>
);

CommunityField.propTypes = {
  input: PropTypes.object,
  meta: PropTypes.object,
  label: PropTypes.string,
  tip: PropTypes.string,
  disabled: PropTypes.bool,
};

export default React.memo(CommunityField);
