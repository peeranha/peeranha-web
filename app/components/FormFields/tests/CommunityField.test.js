import { CommunityField } from '../CommunityField';

const props = {
  input: {},
  meta: {},
  label: 'label',
  tip: 'tip',
  disabled: false,
};

describe('CommunityField', () => {
  it('test', () => {
    expect(CommunityField(props)).toMatchSnapshot();
  });
});
