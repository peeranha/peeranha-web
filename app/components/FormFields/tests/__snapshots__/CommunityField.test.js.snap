// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityField test 1`] = `
<UNDEFINED
  label="label"
  meta={Object {}}
  tip="tip"
>
  <Connect(CommunitySelector)
    Button={[Function]}
    disabled={false}
    input={Object {}}
    selectedCommunityId={0}
  />
</UNDEFINED>
`;
